src/mesido/esdl/profile_parser.py

```diff
@@ -301,18 +301,18 @@ def _load_profiles_from_source(
                     # For multicommidity work profiles need to be assigned to GenericConsumer, but
                     # not for heat network (this asset_potential_errors is used in grow_workflow)
                     if type(asset) is esdl.GenericConsumer:
-                        # asset_potential_errors
-                        self._asset_potential_errors_identified["heat_demand.type"][asset.name] = (
-                            f"Asset {asset.name}: This asset is currently a GenericConsumer please"
-                            " change it to a HeatingDemand"
+                        POTENTIAL_ERRORS.add_potential_issue(
+                            MesidoAssetIssueType.HEAT_DEMAND_TYPE,
+                            asset.id,
+                            f"Asset named {asset.name}: This asset is currently a GenericConsumer"
+                            " please change it to a HeatingDemand",
                         )
                 except KeyError:
                     raise RuntimeError(
                         f"The asset {profile.field} is of type {type(asset)} which is "
                         f"currently not supported to have a profile to be loaded "
                         f"from the database."
                     )
```

Comment:
- Sorry I do not understand. In lines 304-307 we are adding the potential error ->>HEAT_DEMAND_TYPE only. This code will create a list for all the demand with this issue without exiting due to this. This will only cause the code to stop when the exception is raised POTENTIAL_ERRORS.convert_to_exception(...) in grow_workflow.py lines 207-226.

- The section of code that you are commenting on (line 311-315) is another type of error (asset does not cater for a profile or something like this) that we need to add to the potential error list in the future.  Not currently the goal of this PR.

Reply:
Done. Has been added in the grow_workflow with a test case